### PR TITLE
Proper coverage support in CMake and Jenkinsfile.

### PR DESCRIPTION
**Pull request by tobsan**
The coverage macro was somewhat broken and has been fixed. The coverage
step was also added to the Jenkinsfile, and its results are published.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -73,11 +73,15 @@ node {
         }
 
         stage('UnitTest') {
-            runInVagrant(workspace, "cd softwarecontainer/build && sudo ./run-tests.py")
+            runInVagrant(workspace, "cd softwarecontainer/build && sudo ./run-tests.py \"*-OutputBuffer\"")
         }
 
-        stage('ServiceTest') {
-            runInVagrant(workspace, "cd softwarecontainer/servicetest && sudo ./run-tests.sh")
+//        stage('ServiceTest') {
```
Comment (frznlogic) [inline Jenkinsfile]: Did you intentionally comment out the servicetest?

**Review by frznlogic — CHANGES_REQUESTED** [review]
As a more general note, I really miss some kind of comments/explanation of the build system at times like this. It's really messy understanding what's from where when you read this without having worked with it for a while. A description of how lcov is invoked/run would be nice. 

I dont think we have that kind of documentation today at all.

File: Jenkinsfile (hunk 1 above)
Comment (tobsan) [reply Jenkinsfile]: Yes, one service test fails so I never reach the coverage step. Will remove the comments before I consider it mergable.

**Comment by tobsan** [comment]
@frznlogic I added some notes in the README.md on coverage and how to run it.

To anyone reviewing: because of currently failing tests (unrelated to this), Jenkins will report errors here. I think this should be pulled in anyway.

**Pull request merged**